pkg/services/libraryelements/database.go

```diff
@@ -293,18 +292,17 @@ func (l *LibraryElementService) getLibraryElements(c context.Context, store db.D
 	err = store.WithDbSession(c, func(session *db.Session) error {
 		builder := db.NewSqlBuilder(cfg, features, store.GetDialect(), recursiveQueriesAreSupported)
 		builder.Write(selectLibraryElementDTOWithMeta)
-		builder.Write(", ? as folder_name ", cmd.FolderName)
+		// builder.Write(", ? as folder_name ", cmd.FolderName)
 		builder.Write(getFromLibraryElementDTOWithMeta(store.GetDialect()))
 		metrics.MFolderIDsServiceCount.WithLabelValues(metrics.LibraryElements).Inc()
-		// nolint:staticcheck
-		writeParamSelectorSQL(&builder, append(params, Pair{"folder_id", cmd.FolderID})...)
-		builder.Write(" UNION ")
-		builder.Write(selectLibraryElementDTOWithMeta)
-		builder.Write(", dashboard.title as folder_name ")
-		builder.Write(getFromLibraryElementDTOWithMeta(store.GetDialect()))
-		builder.Write(" INNER JOIN dashboard AS dashboard on le.folder_id = dashboard.id AND le.folder_id <> 0")
-		writeParamSelectorSQL(&builder, params...)
```

Comment:
Do we need to refactor the params part? If possible to keep, the less we can change, the better.

Reply:
This part was needed when we still had the union (line 295 in the code being changed). We had `SELECT <first select> UNION <second select>`. The where condition was being applied to the second select. It's gone now and we only have `SELECT <first select>`. The where condition for the first select is being set on [line 294](https://github.com/grafana/grafana/pull/99160/files#diff-b2c64a080ce54d961189cd32a724375105e790dbd5dddd414b404ad25e53a0b7L294) and that hasn't changed. Hope that makes sense. This [query comparison](https://github.com/grafana/grafana/pull/99160#issuecomment-2604227133) might help.

Nonetheless, I tried to bring back as much of the original code as possible.

Reply:
Thanks for the explanation!